test/publicReportingAPI.test.js

```diff
@@ -1,252 +1,369 @@
+      const spySetStatus = jest.spyOn(ClientPublicReportingAPI, 'setStatus').mockImplementation(() => {});
 
-    expect(spyAddLog).toHaveBeenCalledWith(expectedAddLogObj);
-  });
-  it('launchError: should call addLaunchLog with ERROR level and specified parameters', function() {
-    const spyAddLog = jest
-      .spyOn(ClientPublicReportingAPI, 'addLaunchLog')
-      .mockImplementation(() => {});
+      PublicReportingAPI.setStatusCancelled();
 
-    const expectedAddLogObj = {
-      log: {
-        level: 'ERROR',
-        message: 'message text',
-        file,
-      },
-    };
+      expect(spySetStatus).toHaveBeenCalledWith(RP_STATUSES.CANCELLED);
+    });
 
-    PublicReportingAPI.launchError('message text', file);
+    it('setStatusInfo: should call setStatus method with "info" status', function() {
+      const spySetStatus = jest.spyOn(ClientPublicReportingAPI, 'setStatus').mockImplementation(() => {});
 
-    expect(spyAddLog).toHaveBeenCalledWith(expectedAddLogObj);
-  });
-  it('launchFatal: should call addLaunchLog with FATAL level and specified parameters', function() {
-    const spyAddLog = jest
-      .spyOn(ClientPublicReportingAPI, 'addLaunchLog')
-      .mockImplementation(() => {});
+      PublicReportingAPI.setStatusInfo();
 
-    const expectedAddLogObj = {
-      log: {
-        level: 'FATAL',
-        message: 'message text',
-        file,
-      },
-    };
+      expect(spySetStatus).toHaveBeenCalledWith(RP_STATUSES.INFO);
+    });
+
+    it('setStatusWarn: should call setStatus method with "warn" status', function() {
+      const spySetStatus = jest.spyOn(ClientPublicReportingAPI, 'setStatus').mockImplementation(() => {});
+
+      PublicReportingAPI.setStatusWarn();
+
+      expect(spySetStatus).toHaveBeenCalledWith(RP_STATUSES.WARN);
+    });
+
+    it('setLaunchStatusPassed: should call setLaunchStatus method with "passed" status', function() {
+      const spyLaunchStatus = jest.spyOn(ClientPublicReportingAPI, 'setLaunchStatus').mockImplementation(() => {});
+
+      PublicReportingAPI.setLaunchStatusPassed();
+
+      expect(spyLaunchStatus).toHaveBeenCalledWith(RP_STATUSES.PASSED);
+    });
+
+    it('setLaunchStatusFailed: should call setLaunchStatus method with "failed" status', function() {
+      const spyLaunchStatus = jest.spyOn(ClientPublicReportingAPI, 'setLaunchStatus').mockImplementation(() => {});
+
+      PublicReportingAPI.setLaunchStatusFailed();
+
+      expect(spyLaunchStatus).toHaveBeenCalledWith(RP_STATUSES.FAILED);
+    });
+
+    it('setLaunchStatusSkipped: should call setLaunchStatus method with "skipped" status', function() {
+      const spyLaunchStatus = jest.spyOn(ClientPublicReportingAPI, 'setLaunchStatus').mockImplementation(() => {});
+
+      PublicReportingAPI.setLaunchStatusSkipped();
+
+      expect(spyLaunchStatus).toHaveBeenCalledWith(RP_STATUSES.SKIPPED);
+    });
+
+    it('setLaunchStatusStopped: should call setLaunchStatus method with "stopped" status', function() {
```

Comment:
It may be worth combining methods (eg setLaunchStatus, setStatus, addLaunchLog, addLog) and statuses into an array and run these tests using `forEach` method

Reply:
Done!